examples/se2_localization_liekf.cpp

```diff
@@ -0,0 +1,285 @@
+
+    // Declare some temporaries
+    Vector2d                e, z;   // expectation, innovation
+    Matrix2d                E, Z;   // covariances of the above
+    Matrix<double, 3, 2>    K;      // Kalman gain
+    manif::SE2Tangentd      dx;     // optimal update step, or error-state
+
+    //
+    //
+    // CONFIGURATION DONE
+
+
+
+    // DEBUG
+    cout << std::fixed   << std::setprecision(3) << std::showpos << endl;
+    cout << "X STATE     :    X      Y    THETA" << endl;
+    cout << "----------------------------------" << endl;
+    cout << "X initial   : " << X_simulation.log().coeffs().transpose() << endl;
+    cout << "----------------------------------" << endl;
+    // END DEBUG
+
+
+
+
+    // START TEMPORAL LOOP
+    //
+    //
+
+    // Make 10 steps. Measure one GPS position each time.
+    for (int t = 0; t < 10; t++)
+    {
+        //// I. Simulation ###############################################################################
+
+        /// simulate noise
+        u_noise = u_sigmas * Array3d::Random();             // control noise
+        u_noisy = u_nom + u_noise;                          // noisy control
+
+        u_simu   = u_nom;
+        u_est    = u_noisy;
+        u_unfilt = u_noisy;
+
+        /// first we move - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+        X_simulation = X_simulation + u_simu;               // overloaded X.rplus(u) = X * exp(u)
+
+        /// then we receive noisy gps measurement - - - - - - - - - - - - - - - -
+        y_noise = y_sigmas * Array2d::Random();             // simulate measurement noise
+
+        y = X_simulation.translation();                     // position measurement, before adding noise
+        y = y + y_noise;                                    // position measurement, noisy
+
+
+
+
+        //// II. Estimation ###############################################################################
+
+        /// First we move - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+        X = X.plus(u_est, J_x, J_u);                        // X * exp(u), with Jacobians
+
+        P = J_x * P * J_x.transpose() + J_u * U * J_u.transpose();
+
+
+        /// Then we correct using the gps position - - - - - - - - - - - - - - -
+
+        // transform covariance from right to left invariant
+        AdX = X.adj();
+        AdXinv = X.inverse().adj();
+        P_L = AdX * P * AdX.transpose();                    // from eq. (54) in the paper
+
+        // expectation
+        e = X.translation();                                // e = t, for X = (R,t).
+        H.topLeftCorner<2, 2>() = Matrix2d::Identity();
+        H.topRightCorner<2, 1>() = manif::skew(1.0) * X.translation();
+        E = H * P_L * H.transpose();
+
+        // innovation
+        z = y - e;
+        Z = E + R;
+
+        // Kalman gain
+        K = P_L * H.transpose() * Z.inverse();
+
+        // Correction step
+        dx = K * z;
+
+        // Update
+        X = X.lplus(dx);
```

Comment:
I believe this should be `X.lplus(-dx)`.

Reply:
Please see my comment here: https://github.com/artivis/manif/pull/308#discussion_r1773957638